### ISSUE TITLE
VS Code Insiders: Restart Current Session doesn't remove old session

### ISSUE THREAD

**Issue by markm77** [PowerShell/vscode-powershell#2687]
Just letting you people know that with VS Code Insiders using either the normal or preview PowerShell plug-in choosing "PowerShell: Restart Current Session" starts a new session without removing the old one - i.e. you end up with two integrated consoles.

BR,
Mark

Environment Information
=====

Visual Studio Code
-----

| Name | Version |
| --- | --- |
| Operating System | Windows_NT x64 10.0.18363 |
| VSCode | 1.46.0-insider|
| PowerShell Extension Version | 2020.4.3 |

PowerShell Information
-----

|Name|Value|
|---|---|
|PSVersion|7.0.0|
|PSEdition|Core|
|GitCommitId|7.0.0|
|OS|Microsoft Windows 10.0.18363|
|Platform|Win32NT|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.10032.0 6.0.0 6.1.0 6.2.0 7.0.0|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|
|WSManStackVersion|3.0|

Visual Studio Code Extensions
-----

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|applescript|idleberg|0.18.2|
|bash-debug|rogalmic|0.3.8|
|csharp|ms-dotnettools|1.21.18|
|gitignore|michelemelluso|1.0.1|
|gitlens|eamodio|10.2.1|
|openwithcode|kevinchatham|1.0.4|
|powershell-preview|ms-vscode|2020.4.3|
|python|ms-python|2020.4.76186|
|remote-ssh|ms-vscode-remote|0.51.0|
|remote-ssh-edit|ms-vscode-remote|0.51.0|
|swagger-viewer|Arjun|3.0.1|
|vscode-eslint|dbaeumer|2.1.5|
|vscode-jest-test-adapter|kavod-io|0.3.4|
|vscode-js-annotations|lannonbr|0.11.0|
|vscode-pull-request-github|GitHub|0.16.0|
|vscode-test-explorer|hbenl|2.18.0|
|vscode-yaml|redhat|0.7.2|

**Comment by SydneyhSmith** [comment]
Thanks @markm77 for opening this issue, unfortunately we are having some trouble reproducing this...it would be helpful if you could provide a gif or screenshot of the issue

**Comment by markm77** [comment]
Hi @SydneyhSmith ,

Okay sure, here's a GIF showing session re-start with VS Code (not Insiders):
![VS Code](https://user-images.githubusercontent.com/26673915/81756446-afefb300-94b3-11ea-9b53-acfde53633f2.gif)

And here's the same operation with Insiders (you can see in the bottom drop-down that the old Integrated Console is still there):
![VS Code Insiders](https://user-images.githubusercontent.com/26673915/81756435-a9613b80-94b3-11ea-822a-dcdaffb676e3.gif)

Does this help? These are my first GIFs so hope they are okay.

**Comment by SydneyhSmith** [comment]
@markm77 thanks those are great and do help! Is your installation of VSCode a user install or system? It would also be interesting to see what happens when you select the other Integrated Console?

**Comment by rjmholt** [comment]
Just to trace through the code.

The restart command is here:

https://github.com/PowerShell/vscode-powershell/blob/939379d13c1fbc1e1ce87cbc46606ba02e2b314d/src/session.ts#L243-L246

That runs `stop()` on the original process:

https://github.com/PowerShell/vscode-powershell/blob/939379d13c1fbc1e1ce87cbc46606ba02e2b314d/src/session.ts#L212-L241

That in turn disposes the terminal:

https://github.com/PowerShell/vscode-powershell/blob/939379d13c1fbc1e1ce87cbc46606ba02e2b314d/src/process.ts#L146-L161

**Comment by markm77** [comment]
> thanks those are great and do help! Is your installation of VSCode a user install or system? It would also be interesting to see what happens when you select the other Integrated Console?

No problem. I can't recall if I have a system or user install - is there a best way to check? I remember changing this at one point to solve something else. The old Integrated Console works fine but I usually trash-can it as the whole reason for doing a re-start is to unload DLLs.

**Comment by corbob** [comment]
~~Not sure if this command will definitely work, but on my system if I run it it gives me the path of Code...~~

```powershell
get-process -pid $pid | select -expand parent | select path
```

~~If the path is under `C:\Users\` then it's a User install, if it's under `C:\Program Files\` then it's system. Another usual tell is if you get prompted to elevate with UAC for an update then it's System install.~~

If you go to Help > About it should tell you if it's a user setup of system setup:

![image](https://user-images.githubusercontent.com/30301021/82004092-eb56c280-9616-11ea-9c74-80c1bb6f7953.png)

**Comment by markm77** [comment]
Thanks @corbob , both of my installs are System Setup. If you still can't reproduce I'm happy to re-install as User Setup and re-test.

[Context: I remember now that I switched to System Setup as connection to my Windows machine via remote desktop for Mac did not work with normal Microsoft account - [fix](https://answers.microsoft.com/en-us/windows/forum/all/microsoft-remote-desktop-the-user-account-did-not/2bc32545-3fb8-4e1e-8dda-7251114432c7) was to create another local admin user to accept RDP connections. So I'd prefer to keep using the System Setup.]

**Comment by SydneyhSmith** [comment]
@markm77 it would be great if you could try the user setup as a data point for us since we are unable to repro and that is the recommendation from VSCode

**Comment by markm77** [comment]
Hi @SydneyhSmith , I opened up VS Code Insiders today and updated it (as I usually do). And, voila.... the problem is gone. I did update the PowerShell extension the other day and I'm pretty sure I saw the problem after that so I believe something in update to VS Code Insiders has fixed the issue. I've also held off on the 7.0.1 PowerShell update so it wasn't that (I try to update one thing at a time to isolate causes for any issues).

So I will close this issue now and thanks to all for the engagement and help which I really appreciated - gave me confidence as a user! BR, Mark
